apis/externalsecrets/v1beta1/secretstore_gitlab_types.go

```diff
@@ -29,6 +29,12 @@ type GitlabProvider struct {
 	// ProjectID specifies a project where secrets are located.
 	ProjectID string `json:"projectID,omitempty"`
```

Comment:
I think we should make the `projectID` optional. I would treat a "project" or "group" simply as a bucket that holds key/value pairs (with these specific inheritance shenanigans form GitLab)

Reply:
This is already optional. Currently the validation simply expects either projectId or groupdId(s)